client/src/app/pages/applications/components/application-form/application-form-modal.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from "react";
+import { Button, ButtonVariant, Modal } from "@patternfly/react-core";
+import { ApplicationForm, useApplicationFormHook } from "./application-form";
+import { Application } from "@app/api/models";
+import { useTranslation } from "react-i18next";
+
+export interface ApplicationFormModalProps {
+  application: Application | null;
+  onClose: () => void;
+}
+
+export const ApplicationFormModal: React.FC<ApplicationFormModalProps> = ({
+  application,
+  onClose,
+}) => {
+  const { t } = useTranslation();
+  const formProps = useApplicationFormHook({ application, onClose });
+  return (
+    <Modal
+      title={
+        application
+          ? t("dialog.title.updateApplication")
+          : t("dialog.title.newApplication")
+      }
+      variant="medium"
+      isOpen={true}
+      onClose={onClose}
+      actions={[
+        <Button
+          key="submit"
+          id="submit"
+          aria-label="submit"
+          variant={ButtonVariant.primary}
+          isDisabled={formProps.isSubmitDisabled}
+          onClick={formProps.onSubmit()}
+        >
+          {!application ? t("actions.create") : t("actions.save")}
+        </Button>,
+        <Button
+          key="cancel"
+          id="cancel"
+          aria-label="cancel"
+          variant={ButtonVariant.link}
+          isDisabled={formProps.isCancelDisabled}
+          onClick={onClose}
+        >
+          {t("actions.cancel")}
+        </Button>,
+      ]}
```

Comment:
If you want to keep the buttons with the Form itself, you could use a portal to a footer element.  This would follow the general pattern of setting a Modal's render container or some of the toolbar filter stuff.

Something like:
  - Setup a useRef() to a footer container
  - Pass that ref as the render container for the form buttons to the form component
  - In the form component, if a button container is passed, `createPortal` to it, otherwise render in place
  - "Bob's your uncle"

Reply:
@sjd78 @ibolton336 
hmm ...portals...  we could use them but this would limit the functionality  i.e. what if we would like to trigger submit from 2 places? or from some menu, dropdown,, etc ?
But if you think it's the way to go then I can refactor the code in that direction.

Reply:
The relationship between the hook and the form component doesn't need to change.  Using a portal would just let the buttons live inside the form component but be rendered elsewhere.